daml/Marketplace/Issuance/AssetDescription.daml

```diff
@@ -55,11 +55,16 @@ template Index
     maintainer key._1
 
     controller descriptionSignatories can
-      LookupOrInsert : (ContractId Index, Version) with claims : Claims
+      LookupOrInsert : (ContractId Index, Version, Optional (ContractId AssetDescription)) with
```

Comment:
~~I don't understand why you're changing this, if you're not using the return value, up on line 117.~~ Ok so this is to create the post-lifecycle `AssetDescription` version?
I don't think it's a good idea to hide it in here; it becomes a side-effect you need to know as a programmer.

Also beware that we may change this so that we have one `Index` per `AssetLabel` / `Version`, instead of having them all in a  single `Map`.

Reply:
> Ok so this is to create the post-lifecycle AssetDescription version?
Yes.

> I don't understand why you're changing this, if you're not using the return value, up on line 117. 
Yes, probably the return value is not even needed so I could just change back.

> I don't think it's a good idea to hide it in here; it becomes a side-effect you need to know as a programmer.
That's true, where would you do it, in Lifecycle/Process?

Reply:
Probably in Process. But see my general comment about #291

Reply:
See now.